ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -301,7 +301,7 @@ async def wrapped_blocked_send(message):
                 # Do not block right here. Wait for route to be resolved in starlette/patch.py
                 return await self.app(scope, receive, wrapped_send)
             except BlockingException as e:
-                core.set_item(HTTP_REQUEST_BLOCKED, e.args[0])
+                set_blocked(e.args[0])
```

Comment:
Why is this one replaced with `set_blocked()` while the other two were removed?

Reply:
Because the others were already using normal block mechanism that use an equivalent for set_blocked at a different place in the code, while this one is for rasp and needs to set_blocked again in case we are exiting nested contexts.